### ISSUE TITLE
While upgrading using docker-compose setup: Fatal error: Uncaught Error: Call to a member function connection() on null in /var/www/app/vendor/laravel/framework/src/Illuminate/Support/Facades/Schema

### ISSUE THREAD

**Issue by vogelfreiheit** [invoiceninja/dockerfiles#184]
# Logs

```
invoiceninja-app        | Fatal error: Uncaught Error: Call to a member function connection() on null in /var/www/app/vendor/laravel/framework/src/Illuminate/Support/Facades/Schema.php:41
invoiceninja-app        | Stack trace:
invoiceninja-app        | #0 /var/www/app/vendor/laravel/framework/src/Illuminate/Support/Facades/Facade.php(166): Illuminate\Support\Facades\Schema::getFacadeAccessor()
invoiceninja-app        | #1 /var/www/app/vendor/laravel/framework/src/Illuminate/Support/Facades/Facade.php(255): Illuminate\Support\Facades\Facade::getFacadeRoot()
invoiceninja-app        | #2 /var/www/app/app/Exceptions/Handler.php(60): Illuminate\Support\Facades\Facade::__callStatic('hasTable', Array)
invoiceninja-app        | #3 /var/www/app/vendor/laravel/framework/src/Illuminate/Foundation/Bootstrap/HandleExceptions.php(92): App\Exceptions\Handler->report(Object(Symfony\Component\Debug\Exception\FatalThrowableError))
invoiceninja-app        | #4 [internal function]: Illuminate\Foundation\Bootstrap\HandleExceptions->handleException(Object(Symfony\Component\Debug\Exception\FatalThrowableError))
invoiceninja-app        | #5 {main}
invoiceninja-app        |   thrown in /var/www/app/vendor/laravel/framework/src/Illuminate/Support/Facades/Schema.php on line 41
invoiceninja-app        |
invoiceninja-app        | Fatal error: Uncaught Error: Call to a member function connection() on null in /var/www/app/vendor/laravel/framework/src/Illuminate/Support/Facades/Schema.php:41
invoiceninja-app        | Stack trace:
invoiceninja-app        | #0 /var/www/app/vendor/laravel/framework/src/Illuminate/Support/Facades/Facade.php(166): Illuminate\Support\Facades\Schema::getFacadeAccessor()
invoiceninja-app        | #1 /var/www/app/vendor/laravel/framework/src/Illuminate/Support/Facades/Facade.php(255): Illuminate\Support\Facades\Facade::getFacadeRoot()
invoiceninja-app        | #2 /var/www/app/app/Exceptions/Handler.php(60): Illuminate\Support\Facades\Facade::__callStatic('hasTable', Array)
invoiceninja-app        | #3 /var/www/app/vendor/laravel/framework/src/Illuminate/Foundation/Bootstrap/HandleExceptions.php(92): App\Exceptions\Handler->report(Object(Symfony\Component\Debug\Exception\FatalErrorException))
invoiceninja-app        | #4 /var/www/app/vendor/laravel/framework/src/Illuminate/Foundation/Bootstrap/HandleExceptions.php(134): Illuminate\Foundation\Bootstrap\HandleExceptions->handleException(Object(Symfony\Component\Debug\Exception\FatalErrorException))
invoiceninja-app        | #5 [internal function]: Illuminat in /var/www/app/vendor/laravel/framework/src/Illuminate/Support/Facades/Schema.php on line 41

```

# Setup information

Using docker-compose. File is the same as upstream, only change is I use nginx instead of Caddy, and mariadb instead of the traditional MySQL image. It worked just fine before attempting to upgrade to the latest image.

It seems the connection information is not being populated.

**Comment by vogelfreiheit** [comment]
After checking, the .env file is never updated and it is populated with default data....

I can see that my env variables are properly present in the container, though. (using docker inspect)

**Comment by codedge** [comment]
Glad it worked.

Quick fix for the cron container:

```yml
cron:
    image: invoiceninja/invoiceninja:alpine-4
    volumes:
      -  storage:/var/www/app/storage
      -  logo:/var/www/app/public/logo
      -  public:/var/www/app/public
    entrypoint: |
      /bin/sh -c 'sh -s <<EOF
      trap "break;exit" SIGHUP SIGINT SIGTERM
      sleep 300s
      while /bin/true; do
        ./artisan ninja:send-invoices
        ./artisan ninja:send-reminders
        sleep 1d
      done
      EOF'
    networks:
      - invoiceninja
```

**Comment by codedge** [comment]
I am unable to help if you do not provide your full `docker-compose.yml`. There are guides how to run IN 4 or 5 with different web servers in the [wiki](https://github.com/invoiceninja/dockerfiles/wiki).

**Comment by codedge** [comment]
@vogelfreiheit Any updates here? Were you able to solve the problem?

**Comment by vogelfreiheit** [comment]
> @vogelfreiheit Any updates here? Were you able to solve the problem?

Not quite, the move to Alpine has broken phantomjs since it will not run without Glibc. I use the cloud version now as a workaround since playing around library files introduces a bunch of issues (broken ABIs, security problems -you cannot reliably update the DSOs and other dynamic library objects).

I think the move to the alpine base images has been poorly planned and backwards support for the Debian images should never have been removed. This happened between 4.5.18 and 4.5.19 as far as I can see from reviewing the tags.

**Comment by vogelfreiheit** [comment]
On a different note: A proper guide for upgrading to 5.x would be appreciated, including the ability to import/export data.

**Comment by codedge** [comment]
Yes, a guide for updating to 5.x is missing currently. For importing/exporting data a good place to ask is the [Invoice Ninja Slack channel](http://slack.invoiceninja.com/).

**Comment by jdreinhardt** [comment]
@vogelfreiheit I was also having this issue on fresh install of 4.5.19. What I ended up finding is that a few fields in the .env do not update properly from the setup page and need to be added as environment variables to get around it. The one likely causing your issue is likely the DB_HOST is resetting to `localhost` and preventing database login. If you are using the docker-compose then add `- DB_HOST=db` to the app environment section.

I also found a couple other environmental variables that are similar and don't stick without be declared in the docker-compose. They are `MAIL_USERNAME` and `MAIL_PASSWORD` the rest of the mail variables seem to work properly.
